activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/ProcessVariablesHelper.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ *
+ */
+
+package org.activiti.cloud.services.rest.controllers;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.activiti.api.process.model.payloads.SetProcessVariablesPayload;
+import org.activiti.api.process.model.payloads.StartProcessPayload;
+import org.activiti.engine.ActivitiException;
+import org.activiti.spring.process.model.Extension;
+import org.activiti.spring.process.model.ProcessExtensionModel;
+import org.activiti.spring.process.model.VariableDefinition;
+import org.activiti.spring.process.variable.VariableValidationService;
+
+public class ProcessVariablesHelper  {
+   
+    private final Map<String, ProcessExtensionModel> processExtensionModelMap;
+    private final VariableValidationService variableValidationService;
+    private final DateFormatterProvider dateFormatterProvider;
+ 
+    public ProcessVariablesHelper(DateFormatterProvider dateFormatterProvider,
+                                  Map<String, ProcessExtensionModel> processExtensionModelMap,
+                                  VariableValidationService variableValidationService) {
+        this.dateFormatterProvider = dateFormatterProvider;
+        this.processExtensionModelMap = processExtensionModelMap;
+        this.variableValidationService = variableValidationService;
+    }
+    
+    private Optional<Map<String, VariableDefinition>> getVariableDefinitionMap(String processDefinitionKey) {
+        ProcessExtensionModel processExtensionModel = processExtensionModelMap.get(processDefinitionKey);
+
+        return Optional.ofNullable(processExtensionModel)
+                .map(ProcessExtensionModel::getExtensions)
+                .map(Extension::getProperties);
+    }
+    
+    public List<ActivitiException> checkPayloadVariables(Map<String, Object> variablePayloadMap,
+                                                         String processDefinitionKey,
+                                                         boolean validate) {
+        
+        final String errorMessage = "Variable with name {0} does not exists.";
+        final String errorDateTimeParse = "Error parsing date/time variable with a name {0}: {1}";
+        
+        final Optional<Map<String, VariableDefinition>> variableDefinitionMap = getVariableDefinitionMap(processDefinitionKey);
+        List<ActivitiException> activitiExceptions = new ArrayList<>();
+        
+        if (variableDefinitionMap.isPresent()) {
+                    
+            for (Map.Entry<String, Object> var : variablePayloadMap.entrySet()) {
+                String name = var.getKey();
+                Object value = var.getValue();
+                boolean found = false;
+                for (Map.Entry<String, VariableDefinition> entry : variableDefinitionMap.get().entrySet()) {
```

Comment:
Using meaningful names improves code readability. For instance, `payloadVar` instead of `var`, `variableDefinitionEntry`instead of `entry`.

Reply:
Changed